E2_ESRGAN/lib/dataset.py

```diff
@@ -1,44 +1,114 @@
 import io
 from absl import logging
+from functools import partial
 import tensorflow as tf
 import tensorflow_datasets as tfds
 
 
-def scale_down(method="bicubic", dimension=1024, factor=4):
+def scale_down(method="bicubic", dimension=256, factor=4):
   def scale_fn(image, *args, **kwargs):
-    high_resolution = tf.image.resize_with_crop_or_pad(
-        image, dimension, dimension)
+    high_resolution = image
+    if not kwargs.get("no_random_crop", None):
+      high_resolution = tf.image.random_crop(
+          image, [dimension, dimension, image.shape[-1]])
+
     low_resolution = tf.image.resize(
-        image, [dimension // factor, dimension // factor], method=method)
-    return (low_resolution, high_resolution)
+        high_resolution,
+        [dimension // factor, dimension // factor],
+        method=method)
+    low_resolution = tf.clip_by_value(low_resolution, 0, 255)
+    high_resolution = tf.clip_by_value(high_resolution, 0, 255)
+    return low_resolution, high_resolution
+  scale_fn.dimension = dimension
   return scale_fn
 
 
+def augment_image(
+        low_res_map_fn,
+        brightness_delta=0.05,
+        contrast_factor=[0.7, 1.3],
+        saturation=[0.6, 1.6]):
+  """ helper function used for augmentation of images in the dataset. """
+  def augment_fn(low_resolution, high_resolution, *args, **kwargs):
+
+    if not tf.random.uniform(minval=0, maxval=5, shape=[], dtype=tf.int32):
+      # choosing integers from an uniform distribution on [0, 5)
+      # returns 0 with a probabilty of 20% (1 / (5 - 0) = 1/5)
+      # Roughly returning 20% of data without applying augmentation
+      return low_resolution, high_resolution
+
+    high_resolution = tf.image.rot90(
```

Comment:
All these are still being applied to 80% of the dataset. Instead, apply each 'block' of augmentation (reorientations, contrast, brightness, saturation) with ~0.5% probability (where the probability is re-generated for each block).

Reply:
okay!